DigitalLearningSolutions.Web.Tests/Controllers/Login/LoginControllerTests.cs

```diff
@@ -0,0 +1,153 @@
+﻿namespace DigitalLearningSolutions.Web.Tests.Controllers.Login
+{
+    using System.Collections.Generic;
+    using DigitalLearningSolutions.Data.Models.User;
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Data.Tests.Helpers;
+    using DigitalLearningSolutions.Web.Controllers;
+    using DigitalLearningSolutions.Web.Tests.TestHelpers;
+    using DigitalLearningSolutions.Web.ViewModels.Login;
+    using FakeItEasy;
+    using FluentAssertions.AspNetCore.Mvc;
+    using NUnit.Framework;
+
+    internal class LoginControllerTests
+    {
+        private LoginController controller;
+        private ICryptoService cryptoService;
+        private IUserService userService;
+
+        [SetUp]
+        public void SetUp()
+        {
+            userService = A.Fake<IUserService>();
+            cryptoService = A.Fake<ICryptoService>();
+
+            A.CallTo(() => userService.GetAdminUserByUsername(A<string>._))
+                .Returns(UserTestHelper.GetDefaultAdminUser());
+            A.CallTo(() => userService.GetDelegateUsersByUsername(A<string>._))
+                .Returns(new List<DelegateUser> { UserTestHelper.GetDefaultDelegateUser() });
+            A.CallTo(() => cryptoService.VerifyHashedPassword(A<string>._, A<string>._))
+                .Returns(true);
+
+            controller = LoginTestHelper.GetLoginControllerWithUnauthenticatedUser(userService, cryptoService);
+        }
+
+        [Test]
+        public void Index_should_render_if_user_is_unauthenticated()
+        {
+            // When
+            var result = controller.Index();
+
+            // Then
+            result.Should().BeViewResult().WithDefaultViewName();
+        }
+
+        [Test]
+        public void Index_should_redirect_if_user_is_authenticated()
+        {
+            // Given
+            var controllerWithAuthenticatedUser =
+                LoginTestHelper.GetLoginControllerWithAuthenticatedUser(userService, cryptoService);
```

Comment:
Ah, I think it might be nicer to add the authentication case to Integration tests rather than here (sorry for not noticing before). We will have auth related stuff there anyway because we can't test the [Authorize] annotations otherwise, so grouping the auth tests together makes sense and I think will make these unit tests tidier too! :)